explorer/src/components/account/AnchorAccountCard.tsx

```diff
@@ -0,0 +1,176 @@
+import { BorshAccountsCoder } from "@project-serum/anchor";
+import { capitalizeFirstLetter } from "utils/anchor";
+import { ErrorCard } from "components/common/ErrorCard";
+import { PublicKey } from "@solana/web3.js";
+import BN from "bn.js";
+
+import ReactJson from "react-json-view";
+import { useCluster } from "providers/cluster";
+import { useAnchorProgram } from "providers/anchor";
+
+export function AnchorAccountCard({ account }: { account: Account }) {
+  const { url } = useCluster();
+  const program = useAnchorProgram(
+    account.details?.owner.toString() ?? "",
+    url
+  );
+
+  if (!program) {
+    return (
+      <ErrorCard text={"Could not decode anchor program for this address"} />
+    );
+  }
+
+  if (!account.details || !account.details.rawData) {
+    return (
+      <ErrorCard text={"This account is parsed as an SPL-native account"} />
+    );
+  }
+  const accountBuffer = account.details.rawData;
+
+  const discriminator = accountBuffer.slice(0, 8) ?? undefined;
+  if (!discriminator) {
+    return <ErrorCard text={"Failed to find anchor account discriminator"} />;
+  }
+
+  let foundAccountLayoutName: string | undefined;
+  let decodedAnchorAccountData: Object | undefined;
+  if (program) {
+    // Iterate all the structs, see if any of the name-hashes match
+    Object.keys(program.account).forEach((accountType) => {
+      const layoutName = capitalizeFirstLetter(accountType);
+      const discriminatorToCheck =
+        BorshAccountsCoder.accountDiscriminator(layoutName);
+
+      if (discriminatorToCheck.equals(discriminator)) {
+        foundAccountLayoutName = layoutName;
+        const accountDecoder = program.account[accountType];
+        decodedAnchorAccountData = accountDecoder.coder.accounts.decode(
+          layoutName,
+          accountBuffer
+        );
+      }
+    });
+  }
+
+  if (!foundAccountLayoutName || !decodedAnchorAccountData) {
+    return (
+      <ErrorCard
+        text={
+          "Failed to find matching anchor account type for account discriminator"
+        }
+      />
+    );
+  }
+
+  return (
+    <>
+      <div className="card">
+        <div className="card-header">
+          <div className="row align-items-center">
+            <div className="col">
+              <h3 className="card-header-title">{foundAccountLayoutName}</h3>
+            </div>
+          </div>
+        </div>
+
+        <div className="table-responsive mb-0">
+          <table className="table table-sm table-nowrap card-table">
+            <thead>
+              <tr>
+                <th className="w-1 text-muted">Key</th>
+                <th className="text-muted">Value</th>
+              </tr>
+            </thead>
+            <tbody className="list">
+              {decodedAnchorAccountData &&
+                Object.keys(decodedAnchorAccountData).map((key) => (
+                  <AccountRow
+                    key={key}
+                    valueName={key}
+                    // @ts-ignore
```

Comment:
![](https://user-images.githubusercontent.com/13243/161804168-88e9437a-fe7b-4d25-8679-3dcbcf152059.gif)

Reply:
Lol, ts-ignore's fixed in latest commit (requires casting one Object as { [key: string]: any }